### PR TITLE
Minor miscellaneous fixes

### PR DESCRIPTION
**Pull request by killerwife**
Export of my past 2 months of work from vengeance.

### CODE REVIEW

File: src/game/Player.cpp [hunk 1]
```diff
@@ -6930,8 +6930,8 @@ void Player::_ApplyItemBonuses(ItemPrototype const* proto, uint8 slot, bool appl
 
     uint32 armor = proto->Armor;
     // Add armor bonus from ArmorDamageModifier if > 0
-    if (proto->ArmorDamageModifier > 0)
-        armor += uint32(proto->ArmorDamageModifier);
+    /*if (proto->ArmorDamageModifier > 0) This is wrong, the member is used by client to determine bonus armor. proto->Armor already has full armor
+        armor += uint32(proto->ArmorDamageModifier);*/
 
     if (armor)
         HandleStatModifier(UNIT_MOD_ARMOR, BASE_VALUE, float(armor), apply);
```
Comment (boxa) [inline src/game/Player.cpp]: @killerwife
right for WotLK (for TBC without armormod?):

    // If set ScalingStatValue armor get it or use item armor
    uint32 armor = proto->Armor;
    if (ssv)
    {
        if (uint32 ssvarmor = ssv->getArmorMod(proto->ScalingStatValue))
            armor = ssvarmor;
    }
    else if (armor && proto->ArmorDamageModifier > 0.0f)
    {
        if (armor > uint32(proto->ArmorDamageModifier))
            armor = armor - uint32(proto->ArmorDamageModifier);
        else
            armor = 0;
    }

    if (armor)
    {
        UnitModifierType modType = TOTAL_VALUE;
        if (proto->Class == ITEM_CLASS_ARMOR)
        {
            switch (proto->SubClass)
            {
                case ITEM_SUBCLASS_ARMOR_CLOTH:
                case ITEM_SUBCLASS_ARMOR_LEATHER:
                case ITEM_SUBCLASS_ARMOR_MAIL:
                case ITEM_SUBCLASS_ARMOR_PLATE:
                case ITEM_SUBCLASS_ARMOR_SHIELD:
                    modType = BASE_VALUE;
                    break;
            }
        }
        HandleStatModifier(UNIT_MOD_ARMOR, modType, float(armor), apply);
    }

    // Add armor bonus from ArmorDamageModifier if > 0
    if (proto->ArmorDamageModifier > 0.0f)
        HandleStatModifier(UNIT_MOD_ARMOR, TOTAL_VALUE, proto->ArmorDamageModifier, apply);

File: src/game/Unit.cpp [hunk 2]
```diff
@@ -5987,9 +5987,12 @@ int32 Unit::SpellBonusWithCoeffs(SpellEntry const* spellProto, int32 total, int3
         // Spellmod SpellDamage
         if (Player* modOwner = GetSpellModOwner())
         {
-            coeff *= 100.0f;
-            modOwner->ApplySpellMod(spellProto->Id, SPELLMOD_SPELL_BONUS_DAMAGE, coeff);
-            coeff /= 100.0f;
+            if (damagetype != DOT) // DOTs are not affected by these SPELLMODs, at least not in TBC/WOTLK
```
Comment (VladimirMangos) [inline src/game/Unit.cpp]: Maybe switch checks order for performence

Comment (ItsmeAga) [inline src/game/Unit.cpp]: 32383 Empowered corruption got that mod

Comment (killerwife) [inline src/game/Unit.cpp]: Ok I have an idea. Instead do if (coeff) so that we dont add to something thats initially zero.

Comment (ItsmeAga) [inline src/game/Unit.cpp]: coeff > 0 sounds sane, that way hardcoded 0-coeffs in the DB cannot be overwritten either

Comment (VladimirMangos) [inline src/game/Unit.cpp]: @killerwife Ah, you are right )

Comment (killerwife) [inline src/game/Unit.cpp]: Ok, its a flat % mod :P

**Comment by cyberium** [comment]
interresting ones merged thanks

**Pull request closed without merging**